de.dlr.sc.virsat.model.extension.tests.test/src/de/dlr/sc/virsat/model/concept/types/structural/level/HierarchyLevelCheckerTest.java

```diff
@@ -0,0 +1,362 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.concept.types.structural.level;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.model.concept.types.structural.IBeanStructuralElementInstance;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.extension.tests.model.AConceptTestCase;
+import de.dlr.sc.virsat.model.extension.tests.model.TestStructuralElement;
+
+/**
+ * Test class for {@link HierarchyLevelChecker}
+ */
+public class HierarchyLevelCheckerTest extends AConceptTestCase {
+
+	private Concept concept;
+
+	private IHierarchyLevel a;
+	private IHierarchyLevel b;
+	private IHierarchyLevel c;
+	private IHierarchyLevel an;
+	private IHierarchyLevel bn;
+	private IHierarchyLevel cn;
+	private IHierarchyLevel ao;
+	private IHierarchyLevel bo;
+	private IHierarchyLevel co;
+	private IHierarchyLevel ano;
+
+	@Before
+	public void setup() {
+		concept = loadConceptFromPlugin();
+
+		a = createNameMatchingLevel("a", false, false);

```

Comment:
Please explain the hierarchy here in the test. It is very unclear what this data is supposed to do.

Reply:
This is just setup - we create all levels that we will need in tests (like usually we create all the equipments etc.). I can insert a couple of comments there.